sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/MonitorDomain.java

```diff
@@ -1,44 +1,39 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
 
 package com.azure.monitor.opentelemetry.exporter.implementation.models;
 
 import com.azure.core.annotation.Fluent;
-import com.fasterxml.jackson.annotation.JsonAnyGetter;
-import com.fasterxml.jackson.annotation.JsonAnySetter;
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-
-import java.util.HashMap;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
 /**
  * The abstract common base of all domains.
  */
 @Fluent
-@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
-@JsonSubTypes({
-    @JsonSubTypes.Type(MessageData.class),
-    @JsonSubTypes.Type(MetricsData.class),
-    @JsonSubTypes.Type(RemoteDependencyData.class),
-    @JsonSubTypes.Type(RequestData.class),
-    @JsonSubTypes.Type(TelemetryExceptionData.class)
-})
-public class MonitorDomain {
+public class MonitorDomain implements JsonSerializable<MonitorDomain> {
```

Comment:
@alzimmermsft is there a way to tell the code generator that this is an abstract base class, and have the subclasses implement `JsonSerializable<X>` where `X` is the subclass?

Reply:
> ## Generate autorest code
``` yaml
input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/applicationinsights/data-plane/Monitor.Exporters/preview/v2.1/swagger.json
java: true
use: '@autorest/java@4.1.29'
output-folder: ../
namespace: com.azure.monitor.opentelemetry.exporter
generate-client-interfaces: false
service-interface-as-public: true
license-header: MICROSOFT_MIT_SMALL
add-context-parameter: true
context-client-method-parameter: true
generate-client-as-impl: true
artifact-id: azure-monitor-opentelemetry-exporter
directive:
    - rename-model:
        from: TrackResponse
        to: ExportResult
```

btw, this is our autorest code

Reply:
makes sense, thanks, I don't think this is a high priority for us since we only use deserialization in tests